cli/tests/test_status_command.rs

```diff
@@ -62,3 +62,54 @@ fn test_status_ignored_gitignore() {
     Parent commit: zzzzzzzz 00000000 (empty) (no description set)
     "###);
 }
+
+// See <https://github.com/martinvonz/jj/issues/3108>
+#[test]
+fn test_status_display_rebase_instructions() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+
+    let repo_path = test_env.env_root().join("repo");
+    let conflicted_path = repo_path.join("conflicted.txt");
+
+    // PARENT: Write the initial file
+    std::fs::write(&conflicted_path, "initial contents").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["describe", "--message", "Initial contents"]);
+
+    // CHILD1: New commit on top of <PARENT>
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["new", "--message", "First part of conflicting change"],
+    );
+    std::fs::write(&conflicted_path, "Child 1").unwrap();
+
+    // CHILD2: New commit also on top of <PARENT>
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &[
+            "new",
+            "--message",
+            "Second part of conflicting change",
+            "@-",
+        ],
+    );
+    std::fs::write(&conflicted_path, "Child 2").unwrap();
+
+    // CONFLICT: New commit that is conflicted by merging <CHILD1> and <CHILD2>
+    test_env.jj_cmd_ok(&repo_path, &["new", "--message", "boom", "all:(@-)+"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["status"]);
+
+    insta::assert_snapshot!(stdout, @r###"
+    The working copy is clean
+    There are unresolved conflicts at these paths:
+    conflicted.txt    2-sided conflict
+    To resolve the conflicts, start by updating the current change:
+      jj new mzvwutvlkqwt
```

Comment:
This is a bit confusing if `mzvwutvlkqwt` is the change the user just checked out (it's annoying that we don't have a good verb for what `jj new` does). It's an existing problem, but I think it's made worse by printing it more frequently. Maybe we should only print it if the suggestion to check out a non-parent commit?

Reply:
I think as a first approximation it works here and we should wait for issues or comments on discord to change it.

We could introduce a config or cli param to hide it maybe ? It would show by default and then people would deactive it once they're more used to JJ :)

Reply:
Let's say you have this very simple history:

```
◉  y conflict
│ 
◉  x
```

Let's say you had noticed the conflict in `y` (maybe because of the existing hint). So now you do `jj new y` so you can resolve it. If you run `jj status` at this point, I think this PR is going to make it so we print a hint telling you to resolve the conflicts by doing `jj new y`, right? But since you just did `jj new y`, I think it's confusing to tell the user to do that again. Or did I misunderstand?

Reply:
you're right

I reworked it a little to make it not output the current change id and instead generic advice when it's called from status